src/runner/globals.js

```diff
@@ -0,0 +1,30 @@
+/*
+  This is a file for things we need to declare in the global scope.
+  Use only under the greatest duress.
+*/
+
+const Decimal128 = Decimal.clone();
+
+const wrappedBinaryIdentifier = (a) => {
+  if (!(a instanceof Decimal128 || a instanceof Big)) {
+    throw new SyntaxError("Mixed numeric types are not allowed.")
+  }
+
+  return a;
+};
+
+const wrappedConstructorIdentifier = (a) => {
```

Comment:
This is of course repeating the checks in the transformer, but this allows us to do them only for identifiers and get slightly nicer errors with literals. Otherwise, we'd do it for every constructor call.